packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -162,14 +162,59 @@ export class TransformerResolver implements TransformerResolverProvider {
       slotEntry = [];
     }
 
-    slotEntry.push({
-      requestMappingTemplate,
-      responseMappingTemplate,
-      dataSource,
-    });
+    if (!this.updateSlotIfExists(slotName, requestMappingTemplate, responseMappingTemplate)) {
+      slotEntry.push({
+        requestMappingTemplate,
+        responseMappingTemplate,
+        dataSource,
+      });
+    }
     this.slotMap.set(slotName, slotEntry);
   };
 
+  updateSlotIfExists = (
+    slotName: string,
+    requestMappingTemplate?: MappingTemplateProvider,
+    responseMappingTemplate?: MappingTemplateProvider,
+  ): boolean => {
+    const slotEntires = this.slotMap.get(slotName)!;
+    let slotIndex = 1;
+
+    if (!slotEntires) {
+      return false;
+    }
+
+    for (const slotItem of slotEntires) {
+      const [slotItemRequestMappingTemplate, slotItemResponseMappingTemplate] = [
+        (slotItem.requestMappingTemplate as any)?.name ?? '',
```

Comment:
Is there a specific type we can use here instead of `any`?

Reply:
Unfortunately no. We could alter the class structure we have for storing templates but that would be outside the scope of the change intended here and requires quite a bit of refactoring.